pkg/controller/terraform/terraform_controller.go

```diff
@@ -800,18 +779,55 @@ func (r RunOptions) generateActionConfigMap() *corev1.ConfigMap {
 }
 
 func (r RunOptions) generateRole() *rbacv1.Role {
+	// TODO tighten up default rbac security since all the cm and secret names
+	// can be predicted.
+
+	rules := []rbacv1.PolicyRule{
+		{
+			Verbs:     []string{"*"},
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+		},
+	}
+
+	// When using the Kubernetes backend, allow the operator to create secrets and leases
+	secretsRule := rbacv1.PolicyRule{
+		Verbs:     []string{"*"},
+		APIGroups: []string{""},
+		Resources: []string{"secrets"},
+	}
+	leasesRule := rbacv1.PolicyRule{
+		Verbs:     []string{"*"},
+		APIGroups: []string{"coordination.k8s.io"},
+		Resources: []string{"leases"},
+	}
+	if r.configMapData["backend_override.tf"] != "" {
+		// parse the backennd string the way most people write it
+		// example:
+		// terraform {
+		//   backend "kubernetes" {
+		//     ...
+		//   }
+		// }
+		s := strings.Split(r.configMapData["backend_override.tf"], "\n")
+		for _, line := range s {
+			// Assuming that config lines contain an equal sign
+			// All other lines are discarded
+			if strings.Contains(line, "backend ") && strings.Contains(line, "kubernetes") {
```

Comment:
I guess using the HCL parser here would be overkill?

Reply:
I figured this naive approach would be "good enough." Covers like 80%. If adoption takes off, better solutions will be scoped.